js/pagedtable.js

```diff
@@ -736,131 +786,274 @@ var PagedTable = function (pagedTable, source) {
-    header.style.transitionDelay = "0";
-
-    renderBody(false);
-
-    if (headerOld) {
-      headerOld.style.position = "absolute";
-      headerOld.style.transform = "translateX(0px)";
-      headerOld.style.opacity = "1";
-      headerOld.style.transition = "transform 100ms linear, opacity 100ms";
-      headerOld.setAttribute("class", "pagedtable-remove-head");
-      if (headerOld.style.transitionEnd) {
-        headerOld.addEventListener("transitionend", function() {
-          var headerOldByClass = thead.querySelector(".pagedtable-remove-head");
-          if (headerOldByClass) thead.removeChild(headerOldByClass);
+  me.toggleRow = function(rownum){
+    var row = pagedTable.querySelector(".row_" + rownum);
+    row.style.visibility = row.style.visibility == "visible" ? "collapse" : "visible"
+  }
+  
+  me.styleColumn = function(el_class, styles){
+    var coltostyle = pagedTable.querySelectorAll("." + el_class);
+    for(var s in styles) {
+        coltostyle.forEach(function(e){
+          e.style[s] = styles[s];
         });
+    }    
+  };
+  
+ // to try to prevent double calculation, the columns must be created. 
+  me.makeColumn = function(column_idx, backwards){
+    // add column to header
+    var pt_header = pagedTable.querySelector("thead");
+    var pt_body = pagedTable.querySelector("tbody");
+    
+    // If the column already exists, toggle its visibility, otherwise create it.
+    var colExists = pt_header.querySelector(".col_" + column_idx) != null;
+    if(colExists){
+      me.toggleColumn("col_" + column_idx);
+    }else{
+      
+      var header_element = makeColumnHeaderElement(column_idx);
+      var body_elements = makeColumnBodyElements(column_idx);
+      
+      if(backwards){
+        pt_header.querySelector(".right-navigator-column").after(header_element);
+      } else{     
+        pt_header.querySelector(".left-navigator-column").before(header_element);
       }
-      else {
-        thead.removeChild(headerOld);
-      }
+      
+      body_elements.map(function(el, idxRow){
+        var row = pt_body.querySelector(".row_"+idxRow)
+        
+        if(backwards){
+          row.querySelector(".right-navigator-column").after(el);
+        } else{     
+          row.querySelector(".left-navigator-column").before(el);
+        }
+      })
+      
+      me.styleColumn("col_" + column_idx,{
+        opacity:0,
+        transition: "",
+      })
     }
 
-    if (tbodyOld) table.removeChild(tbodyOld);
-
-    tbody.style.opacity = "0";
-    tbody.style.transition = "transform 200ms linear, opacity 200ms";
-    tbody.style.transitionDelay = "0ms";
-
-    // force relayout
-    window.getComputedStyle(header).opacity;
-    window.getComputedStyle(tbody).opacity;
-
-    if (headerOld) {
-      headerOld.style.transform = backwards ? "translateX(20px)" : "translateX(-30px)";
-      headerOld.style.opacity = "0";
+  };
+  
+  var ForEach = function(array, direction, func){
```

Comment:
I think we want this one named `forEach()` or `forEachDir()` to follow caml naming conventions.